conans/client/migrations_settings.py

```diff
@@ -2413,6 +2414,12 @@
                 <<: *visual_studio
             apple-clang:
                 <<: *apple_clang
+    intel-cc:
```

Comment:
in general, I'd love to see an explanation/motivation on these scheme and naming
e.g., why `classic` flavor is not just a new version of the existing `intel` compiler?
why sub-setting fits better than several new compilers?
also, what about `intel-clang` as new compilers seems to be clang based? do they share many things in common?

Reply:
Let's go step by step and discuss the different points:
* The naming, for sure, we can think of another one that can fit better. This name is easy to read/write so, why not?
* Why different `modes` rather than different compilers? According to [this Intel reference](https://software.intel.com/content/www/us/en/develop/articles/porting-guide-for-icc-users-to-dpcpp-or-icx.html):

```
The following are the guiding principles for ICX:

* ICX and ICC Classic use different compiler drivers. The ICC Classic drivers are icc, icpc, and icl. The ICX drivers are icx and icpx. Use icx to compile and link C programs, and icpx for C++ programs. 
Unlike the icc driver, icx does not use the file extension to determine whether to compile as C or C+. Users must invoke icpx to compile C+ files. In addition to providing a core C++ Compiler, ICX is the base compiler for the Intel® oneAPI Data Parallel C++ Compiler and its new driver, dpcpp.
* For Data Parallel C++ applications: The underlying compiler is ICX and the driver is dpcpp. 
* For Standard C++ applications: Intel recommends using the ICC Classic production compiler and the icc/icpc/icl driver.
```
All the compilers are more or less related between them and they are different for sure, but we can keep it simpler by changing only the mode instead of declaring 3 different ones with the same flags, installation paths, etc.
* They are indeed LLVM-based and, they admit the Clang flags but, in my opinion, all the code related to compilers based on another one is a bit convoluted so I think it's better to avoid it.

I hope it's clearer now. By the way, @SSE4 @memsharded please, share your thoughts here and let me know what you think about this new approach, it would be great 😁